reVX/wind_dirs/prominent_wind_dirs.py

```diff
@@ -0,0 +1,244 @@
+                                                       points)
+
+        directions = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+        neighbor_gids = pd.DataFrame(neighbor_gids,
+                                     columns=directions,
+                                     index=sc_point_gids)
+
+        return neighbor_gids
+
+    def _get_prominent_directions(self, excl_area=0.0081, max_workers=None,
+                                  chunk_point_len=100):
+        """
+        Aggregate power rose data to
+
+        Parameters
+        ----------
+        excl_area : float, optional
+            Area of an exclusion cell (square km), by default 0.0081
+        max_workers : int | None, optional
+            Number of cores to run summary on. None is all
+            available cpus, by default None
+        chunk_point_len : int, optional
+            Number of SC points to process on a single parallel worker,
+            by default 100
+
+        Returns
+        -------
+        prominent_dirs : pandas.DataFrame
+            Update meta data table with neighboring supply curve point gids
+            at each cardinal direction and in order of prominent power rose
+            direction
+        """
+        agg_out = self.aggregate(excl_area=excl_area, max_workers=max_workers,
+                                 chunk_point_len=chunk_point_len)
+
+        meta = agg_out.pop('meta')
+        neighbor_gids = self._get_neighbors(self._excl_fpath,
+                                            meta['sc_point_gids'].values,
+                                            resolution=self._resolution)
+
+        dir_pos = self._map_direction_pos(self._h5_fpath)
+        prominent_dirs = agg_out.pop('powerrose_100m')[dir_pos]
+        prominent_dirs = np.argsort(prominent_dirs.T)
+        prominent_dirs = np.take_along_axis(neighbor_gids.values,
+                                            prominent_dirs,
+                                            axis=1)
+
+        columns = ['prominent_direction_{}'.format(i + 1)
+                   for i in range(len(dir_pos))]
+        prominent_dirs = pd.DataFrame(prominent_dirs,
+                                      index=meta['sc_point_gids'].values,
+                                      columns=columns)
+        prominent_dirs = neighbor_gids.join(prominent_dirs)
+        del neighbor_gids
+        prominent_dirs = meta.join(prominent_dirs)
+
+        return prominent_dirs
+
+    @classmethod
+    def run(cls, power_rose_h5_fpath, excl_fpath,
+            agg_dset='powerrose_100m', tm_dset='techmap_wtk',
+            resolution=64, excl_area=0.0081, max_workers=None,
+            chunk_point_len=100, out_fpath=None):
+        """
+        Aggregate powerrose to supply curve points, find neighboring supply
+        curve point gids and rank them based on prominent powerrose direction
+
+        Parameters
+        ----------
+        power_rose_h5_fpath : str
```

Comment:
This could benefit from some description of the data structure of the power rose dataset. For example, columns are sites and rows are directions? Or vis versa?

Reply:
done, its WTK with directions on axis 0